src/connector/src/sink/mod.rs

```diff
@@ -326,13 +326,30 @@ impl SinkWriterParam {
     }
 }
 
+pub enum DecoupleMode {
+    Force(bool),
+    Recommend(bool),
+}
+
+impl DecoupleMode {
+    pub fn is_decouple(&self) -> bool {
+        matches!(
+            self,
+            DecoupleMode::Force(true) | DecoupleMode::Recommend(true)
+        )
+    }
+}
+
 pub trait Sink: TryFrom<SinkParam, Error = SinkError> {
     const SINK_NAME: &'static str;
     type LogSinker: LogSinker;
     type Coordinator: SinkCommitCoordinator;
 
-    fn default_sink_decouple(_desc: &SinkDesc) -> bool {
-        false
+    /// Infer the decouple mode from the sink properties.
```

Comment:
I think we can directly change the `default_sink_decouple` to  `fn is_sink_decouple(desc: &SinkDesc, user_specified: SinkDecouple) -> bool` and in frontend we can it directly.

For the enum `SinkDecouple` we can implement have a method like `fn is_decouple(&self) -> Option<bool>`, and then the default implementation of `is_sink_decouple` can be `user_specified.is_decouple().unwrap_or(false)`.

Reply:
If my understanding is correct, the `user_specified: SinkDecouple` is the session config.  `fn is_sink_decouple(desc: &SinkDesc, user_specified: SinkDecouple) -> bool` determines whether the sink is decoupled. It also will check if there are conflicts between them. 🤔 Maybe it should be `fn is_sink_decouple(desc: &SinkDesc, user_specified: SinkDecouple) -> Result<bool>`. and return errors when there are conflicts.

I agree with this idea and we can remove the logic here https://github.com/risingwavelabs/risingwave/blob/61c025d1c7d1705c193901e98c57b221c0f9a711/src/frontend/src/optimizer/plan_node/stream_sink.rs#L643.

So that we can determine the decouple logic in one place rather than separate them.